spec/memo_wise_spec.rb

```diff
@@ -357,5 +357,208 @@ def self.class_method; end
+        end
+
+        context "when an invalid hash key is passed to .memo_wise" do
+          let(:module_with_memo) do
+            Module.new do
+              prepend MemoWise
+
+              def self.module_method; end
+            end
+          end
+
+          it "raises an error when passing a key which is not `self:`" do
+            expect do
+              module_with_memo.send(:memo_wise, bad_key: :module_method)
+            end.
+              to raise_error(
+                ArgumentError,
+                "`:self` is the only key allowed in memo_wise"
+              )
+          end
+        end
+      end
+
+      context "when defined with scope 'module << self'" do
+        include_context "with context for module methods via scope "\
+                        "'class << self'"
+
+        # Use the module as the target of "#memo_wise shared examples"
+        let(:target) { module_with_memo }
+
+        it_behaves_like "#memo_wise shared examples"
+
+        it "creates a module-level instance variable" do
+          # NOTE: this test ensure the inverse test above continues to be valid
+          expect(module_with_memo.instance_variables).to include(:@_memo_wise)
+        end
+      end
+    end
+
+    # These test cases would fail due to a JRuby bug
+    # Skipping to make build pass until the bug is fixed
+    # https://github.com/jruby/jruby/issues/6758
+    unless RUBY_PLATFORM == "java"
+      context "with module mixed into other classes" do
+        context "when extended" do
+          context "when defined with 'def'" do
+            include_context "with context for module methods via normal scope"
+
+            before(:each) do
+              stub_const("ModuleWithMemo", module_with_memo)
+            end
+
+            let(:class_extending_module_with_memo) do
+              Class.new do
+                extend ModuleWithMemo
+              end
+            end
+
+            let(:target) { class_extending_module_with_memo }
+
+            it_behaves_like "#memo_wise shared examples"
+          end
+
+          context "when 1 module extended by 2 classes" do
+            let(:module_with_memo) do
+              Module.new do
+                prepend MemoWise
+
+                def test_method
+                  Random.rand
+                end
+                memo_wise :test_method
+              end
+            end
+            let(:class_a_extending_module_with_memo) do
+              Class.new do
+                extend ModuleWithMemo
+              end
+            end
+            let(:class_b_extending_module_with_memo) do
+              Class.new do
+                extend ModuleWithMemo
+              end
+            end
+
+            before(:each) do
+              stub_const("ModuleWithMemo", module_with_memo)
+            end
+
+            it <<~DESC do
+            does memoize class method for both classes
+            with same name with separate caches
+            DESC
```

Comment:
```suggestion
            it "memoizes each extended class separately" do
```

This suggestion would help fix a Rubocop issue